ohhell.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+from collections import defaultdict
+from dataclasses import dataclass
+
+
+DECK_SIZE = 52
+SCORE_MULTIPLIER = 10
+with open("VERSION") as file:
+    GAME_VERSION = file.read()
+
+
+@dataclass
+class Player:
+    name: str
+    score: int = 0
+    bids: int = 0
+    wins: int = 0
+
+
+def enter_players(num_players: int):
+    players = []
+    for i in range(num_players):
+        players.append(
+            Player(str(input("Enter the player's name: ")))
+        )
+    print("=====================================")
+    return players
+
+
+def enter_player_guesses(players: list[Player], round: int):
+    total_bids = 0
+    allowed_bids = round + 1
+    for i in range(len(players)):
+        player = players[(i + round) % len(players)]
+        if i == len(players) - 1 and total_bids <= allowed_bids:
+            print(f"The next player CANNOT GUESS {allowed_bids - total_bids}")
+        player.bids = int(input(f"How many tricks will {player.name} win?: "))
+        total_bids += player.bids
+
+
+def enter_tricks_won(players: list[Player], round: int):
+    for i in range(len(players)):
+        player = players[(i+round) % len(players)]
+        player.wins = int(input(f"How many tricks did {player.name} win?: "))
+
+
+def update_scores(players: list[Player]):
+    for player in players:
+        if player.bids != player.wins:
+            player.score -= abs(player.wins - player.bids) * 10
+        else:
+            player.score += 20 + (10 * player.bids)
+
+    print("Current Scores")
+    for player in sorted(players, key=lambda x:x.score, reverse=True):
+        print(f"{player.name}: {player.score} points")
+
+
+def display_winner(players: list[Player]):
```

Comment:
Nit: Would like to see this showcase everyone's final rank(or maybe the top 3 for now) WDYT? I think it would be great for the user to see this after completing the game.

Reply:
Ya I think this could be a nice touch.